java/ql/lib/semmle/code/java/Annotation.qll

```diff
@@ -99,19 +232,73 @@ private predicate sourceAnnotValue(Annotation a, Method m, Expr val) {
 
 /** An abstract representation of language elements that can be annotated. */
 class Annotatable extends Element {
-  /** Holds if this element has an annotation. */
-  predicate hasAnnotation() { exists(Annotation a | a.getAnnotatedElement() = this) }
+  /** Holds if this element has an annotation, including inherited annotations. */
+  predicate hasAnnotation() { exists(getAnAnnotation()) }
 
-  /** Holds if this element has the specified annotation. */
+  /** Holds if this element has a declared annotation, excluding inherited annotations. */
+  predicate hasDeclaredAnnotation() { exists(getADeclaredAnnotation()) }
+
+  /**
+   * Holds if this element has the specified annotation, including inherited
+   * annotations.
+   */
   predicate hasAnnotation(string package, string name) {
     exists(AnnotationType at | at = this.getAnAnnotation().getType() |
       at.nestedName() = name and at.getPackage().getName() = package
     )
   }
 
-  /** Gets an annotation that applies to this element. */
+  /**
+   * Gets an annotation that applies to this element, including inherited annotations.
+   * The results only include _direct_ annotations; _indirect_ annotations, that is
+   * repeated annotations in an (implicit) container annotation, are not included.
+   */
+  // This predicate is overridden by Class to consider inherited annotations
```

Comment:
Move comment inside predicate to avoid breaking qldoc apart from the documented predicate